Packs/IllusiveNetworks/Integrations/IllusiveNetworks/IllusiveNetworks_test.py

```diff
@@ -0,0 +1,327 @@
+                       {'hostname': 'aaa', 'domainName': 'bbb', 'ipAddresses': '1.1.1.1'}}
+
+
+def test_get_asm_cj_insight_command(requests_mock):
+    mock_response = {'data': [], 'hostname': 'bbb', 'machineTagAndSubTags': {'tag': 'tag', 'subTag': 'sub'}}
+    requests_mock.get('https://server/api/v1/crownjewels/insights', json=mock_response, status_code=202)
+
+    client = Client(base_url='https://server', verify=False, credentials='test')
+
+    args = {
+    }
+    _, outputs, _ = get_asm_cj_insight_command(client, args)
+
+    assert outputs == {'Illusive.AttackSurfaceInsights.CrownJewel(val.hostname == obj.hostname)':
+                       {'data': [], 'hostname': 'bbb', 'machineTagAndSubTags': {'tag': 'tag', 'subTag': 'sub'}}}
+
+
+def test_get_deceptive_users_command(requests_mock):
+    mock_response = {'data': [], 'hostname': 'bbb', 'machineTagAndSubTags': {'tag': 'tag', 'subTag': 'sub'}}
+    requests_mock.get('https://server/api/v1/deceptive-entities/users?deceptive_user_type=ALL', json=mock_response)
+
+    client = Client(base_url='https://server', verify=False, credentials='test')
+    args = {
+    }
+    _, outputs, _ = get_deceptive_users_command(client, args)
+
+    assert outputs == {'Illusive.DeceptiveUser(val.userName == obj.userName)':
+                       {'data': [], 'hostname': 'bbb', 'machineTagAndSubTags': {'tag': 'tag', 'subTag': 'sub'}}}
+
+
+def test_get_deceptive_servers_command(requests_mock):
+    mock_response = {'data': [], 'hostname': 'bbb', 'machineTagAndSubTags': {'tag': 'tag', 'subTag': 'sub'}}
+    requests_mock.get('https://server/api/v1/deceptive-entities/servers?deceptive_server_type=SUGGESTED',
+                      json=mock_response)
+
+    client = Client(base_url='https://server', verify=False, credentials='test')
+    args = {
+        'type': "SUGGESTED",
+    }
+    _, outputs, _ = get_deceptive_servers_command(client, args)
+
+    assert outputs == {'Illusive.DeceptiveServer(val.host == obj.host)':
+                       {'data': [], 'hostname': 'bbb', 'machineTagAndSubTags': {'tag': 'tag', 'subTag': 'sub'}}}
+
+
+def test_get_forensics_timeline_command(requests_mock):
+    mock_response = {'IncidentId': "aaa", 'Status': 'Done', 'Evidence': []}
+    requests_mock.get('https://server/api/v1/forensics/timeline?incident_id=3'
+                      '&end_date=1994-09-24T17:30:00.000Z&start_date=1993-09-24T17:30:00.000Z', json=mock_response)
+
+    client = Client(base_url='https://server', verify=False, credentials='test')
+    args = {
+        'incident_id': "3",
+        'start_date': "1993-09-24T17:30:00.000Z",
+        'end_date': "1994-09-24T17:30:00.000Z"
+    }
+    _, outputs, _ = get_forensics_timeline_command(client, args)
+
+    assert outputs == {'Illusive.Forensics(val.IncidentId == obj.IncidentId)':
+                       {'IncidentId': '3', 'Status': 'Done', 'Evidence':
+                        {'IncidentId': 'aaa', 'Status': 'Done', 'Evidence': []}}}
```

Comment:
why do you have IncidentId inside of IncidentId? Is it like nested incidents?

Reply:
yes